src/main/java/bio/terra/stairway/HookWrapper.java

```diff
@@ -0,0 +1,77 @@
+package bio.terra.stairway;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HookWrapper {
+  private static final Logger logger = LoggerFactory.getLogger(HookWrapper.class);
+  private StairwayHook stairwayHook;
+
+  public HookWrapper(StairwayHook stairwayHook) {
+    this.stairwayHook = stairwayHook;
+  }
+
+  public HookAction startFlight(FlightContext context) {
+    if (stairwayHook == null) {
```

Comment:
Since there is a significant amount of duplicated code between each HookWrapper method and it's more than 1-2 lines, I think it might be worthwhile having a private method that takes the right StairwayHook method instead.

Reply:
That was the original approach that @snf2ye had coded. I pushed back, because the bulk of the private method was a switch statement to decide which hook to call.

The other alternative would be a private method that took the function to call as a parameter. That adds a lot of mechanism to save an if statement and a try-block in my view.

This may be one of those things where there is no way to make it beautiful code.